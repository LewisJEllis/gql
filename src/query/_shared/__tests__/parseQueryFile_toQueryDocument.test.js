/* @flow */
import { toQueryDocument } from '../parseQueryFile';
import { Source } from 'graphql/language/source';

const relayParser = ['EmbeddedQueryParser', { startTag: 'Relay\\.QL`', endTag: '`' }];

test('plain queries', () => {
  const text = `
    fragment test on Viewer {
      image
    }
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: 'QueryParser' },
  );

  expect(qd).toMatchSnapshot();
});

test('works with invalid queries', () => {
  const text = `
    fragment Viewer {
      image
    }
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: 'QueryParser' },
  );

  expect(qd).toMatchSnapshot();
});

test('extract embedded queries', () => {
  const text = `
    fragments: {
      viewer: () => Relay.QL\`
        fragment test on Viewer {
          image
        }
      \`,
    },
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: relayParser, isRelay: true },
  );

  expect(qd).toMatchSnapshot();
});

test('extract embedded queries multiline tag', () => {
  const text = `
    some multiline tag
    \`
      fragment test on Test {
        name
      }
    \`
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    {
      parser: [
        'EmbeddedQueryParser',
        { startTag: 'some multiline tag\\s+`', endTag: '`' },
      ],
    },
  );

  expect(qd).toMatchSnapshot();
});

test('extract multiple embedded queries', () => {
  const text = `
    fragments: {
      viewer: () => Relay.QL\`
        fragment test on Viewer {
          image
        }
      \`,
    },

    fragments: {
      viewer: () => Relay.QL\`
        fragment test on Viewer {
          image
        }
      \`,
    };

    const a = Relay.QL\`
      fragment test on Viewer {
        image
      }
    \`;
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: relayParser, isRelay: true },
  );

  expect(qd).toMatchSnapshot();
});

test('add dummy fragment name if missing (relay)', () => {
  const text = `
    fragments: {
      viewer: () => Relay.QL\`
        fragment on Viewer {
          image
        }
      \`,
    },
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: relayParser, isRelay: true },
  );

  expect(qd).toMatchSnapshot();
});

test('add dummy fragment name if missing (relay) and `on` in next line', () => {
  const text = `
    fragments: {
      viewer: () => Relay.QL\`
        fragment
              on Viewer {
          image
        }
      \`,
    },
  `;

  const qd = toQueryDocument(
    new Source(text, 'query.js'),
    { parser: relayParser, isRelay: true },
  );

  expect(qd).toMatchSnapshot();
});

describe('replace inline js fragment with dummy fragment', () => {
  it('single line', () => {
    const text = `
      fragments: {
        viewer: () => Relay.QL\`
          fragment on Viewer {
            \${component.getFragment('viewer')} name
          }
        \`,
      },
    `;

    const qd = toQueryDocument(
      new Source(text, 'query.js'),
      { parser: relayParser, isRelay: true },
    );

    expect(qd).toMatchSnapshot();
  });

  it('multiline', () => {
    const text = `
      fragments: {
        viewer: () => Relay.QL\`
          fragment on Viewer {
            \${component.getFragment('viewer', {
              var1: variables.var1,
              var2: variables.var2,
            })} name
          }
        \`,
      },
    `;

    const qd = toQueryDocument(
      new Source(text, 'query.js'),
      { parser: relayParser, isRelay: true },
    );

    expect(qd).toMatchSnapshot();
  });
});

