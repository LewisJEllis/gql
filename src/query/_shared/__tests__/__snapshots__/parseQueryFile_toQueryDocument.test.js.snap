exports[`replace inline js fragment with dummy fragment multiline 1`] = `
"
                  
                               
          fragment _ on Viewer {
            ...F,,,,,,,,,,,,,,,,,,,,,,,,,,,,, ,
              ,,,,, ,,,,,,,,,,,,,,,
              ,,,,, ,,,,,,,,,,,,,,,
            ,,, name
          }
          
        
    "
`;

exports[`replace inline js fragment with dummy fragment single line 1`] = `
"
                  
                               
          fragment _ on Viewer {
            ...F,,,,,,,,,,,,,,,,,,,,,,,,,,,,,, name
          }
          
        
    "
`;

exports[`test add dummy fragment name if missing (relay) 1`] = `
"
                
                             
        fragment _ on Viewer {
          image
        }
        
      
  "
`;

exports[`test add dummy fragment name if missing (relay) and \`on\` in next line 1`] = `
"
                
                             
        fragment
              _ on Viewer {
          image
        }
        
      
  "
`;

exports[`test extract embedded queries 1`] = `
"
                
                             
        fragment test on Viewer {
          image
        }
        
      
  "
`;

exports[`test extract embedded queries multiline tag 1`] = `
"
                      
     
      fragment test on Test {
        name
      }
     
  "
`;

exports[`test extract multiple embedded queries 1`] = `
"
                
                             
        fragment test on Viewer {
          image
        }
         
      

                
                             
        fragment test on Viewer {
          image
        }
         
      

                       
      fragment test on Viewer {
        image
      }
      
  "
`;

exports[`test plain queries 1`] = `
"
    fragment test on Viewer {
      image
    }
  "
`;

exports[`test works with invalid queries 1`] = `
"
    fragment Viewer {
      image
    }
  "
`;
